### PR TITLE
Feature/signup

### PR DESCRIPTION
**Pull request by NOXY1**

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+									if(result) {
+										this.setState({redirect: true});
+									}
+								})
+								.catch(error => {
+									this.setState({notification: error, notificationType: 'notification error' });
+								});
+
+		this.showNotification = (notification) => {
+			this.setState({notification});
```
Comment (Gorett) [inline src/components/Form.js]: this.setState({ notification }); Do not forget spaces here.

File: src/components/Form.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+									this.setState({notification: result, notificationType: 'notification success' });
+									if(result) {
+										this.setState({redirect: true});
+									}
+								})
+								.catch(error => {
+									this.setState({notification: error, notificationType: 'notification error' });
+								});
+
+		this.showNotification = (notification) => {
```
Comment (Gorett) [inline src/components/Form.js]: this.showNotification = notification => { You can use this notation if there is only one argument

File: src/components/Form.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+			this.setState({notification});
+
+			setTimeout(() => {
+				this.setState({notification: null})
+			});
```
Comment (Gorett) [inline src/components/Form.js]: setTimeout withous second argument will hide notification on next tick. We need 5 seconds here.

File: src/components/Form.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+								})
+								.catch(error => {
+									this.setState({notification: error, notificationType: 'notification error' });
+								});
+
+		this.showNotification = (notification) => {
+			this.setState({notification});
+
+			setTimeout(() => {
+				this.setState({notification: null})
```
Comment (Gorett) [inline src/components/Form.js]: this.setState({ notification: null }); Do not forget spaces here.

File: src/components/Form.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+		}
+
+		return(
+			<Fragment>
+				<p className={this.state.notificationType}>{this.state.notification}</p>
+				<form className='ml'>
+					<TextField 
+						name='login'
+						label='Login'
+						value={this.state.login}
```
Comment (Gorett) [inline src/components/Form.js]: You already destructed 'login' at the beginning of the render method

File: src/components/Form.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='password'
+						label='Password'
+						value={this.state.password}
```
Comment (Gorett) [inline src/components/Form.js]: You already destructed 'password' at the beginning of the render method

File: src/components/Form.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+						type='password'
+					/>
+					<br />
+					<Button variant='contained'
+							label='Submit'
```
Comment (Gorett) [inline src/components/Form.js]: Change to Sign Up

File: src/components/Home.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React, { Fragment } from 'react';
+import '../App.css';
+
+
+const Home = () => {
```
Comment (Gorett) [inline src/components/Home.js]: can be const Home = () => (<h1 className='home'>Home</h1>)

File: src/components/LogIn.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+                  this.showNotification(error, 'notification error');
+                  // this.setState({notification: error, notificationType: 'notification error' });
+                });
+
+    // setTimeout(this.hideNotification, 5000);
+
+  }
+
+  showNotification = (notification, notificationType, time = 5000) => {
+    this.setState({notification: notification, notificationType: notificationType});
```
Comment (Gorett) [inline src/components/LogIn.js]: this.setState({notification: notification, notificationType: notificationType}); Can be simplified to this.setState({ notification, notificationType })

File: src/components/LogIn.js [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    return(
+      <Fragment>
+        {!!notification && <p className={notificationType}>{notification}</p>}
+        <form className='ml'>
+          <TextField 
+            name='login'
+            label='Login'
+            value={this.state.login}
```
Comment (Gorett) [inline src/components/LogIn.js]: You already destructed 'login' at the beginning of the render method

File: src/components/LogIn.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+            onChange={e => this.handleChangeInput(e)}
+            margin="normal"
+          />
+          <br />
+          <TextField 
+            name='password'
+            label='Password'
+            value={this.state.password}
```
Comment (Gorett) [inline src/components/LogIn.js]: You already destructed 'password' at the beginning of the render method

File: src/components/NotFound.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+import React, { Component, Fragment } from 'react';
+
+
+class NotFound extends Component {
```
Comment (Gorett) [inline src/components/NotFound.js]: Can be simplified to arrow component

File: src/components/SignUp.js [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+			return (<Redirect to={'/home'} />)
+		}
+
+		return(
+			<Fragment>
+				<form className='ml'>
+					<TextField 
+						name='firstName'
+						label='Name'
+						value={this.state.firstName}
```
Comment (Gorett) [inline src/components/SignUp.js]: You already destructed it at the beginning of the render method

File: src/components/SignUp.js [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='lastName'
+						label='Surname'
+						value={this.state.lastName}
```
Comment (Gorett) [inline src/components/SignUp.js]: You already destructed it at the beginning of the render method

File: src/components/SignUp.js [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='email'
+						label='Email'
+						value={this.state.email}
```
Comment (Gorett) [inline src/components/SignUp.js]: You already destructed it at the beginning of the render method

File: src/components/SignUp.js [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='birthday'
+						label='Birthday'
+						value={this.state.birthday}
```
Comment (Gorett) [inline src/components/SignUp.js]: You already destructed it at the beginning of the render method

File: src/components/SignUp.js [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<FormControlLabel
+			          control={
+			            <Checkbox
+			              checked={this.state.checkedB}
```
Comment (Gorett) [inline src/components/SignUp.js]: checkedB like a Melanie c? Use understandable names for variables.

File: src/components/SignUp.js [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+			              value="checkedB"
```
Comment (Gorett) [inline src/components/SignUp.js]: This value is static. Is it correct? Or you want to use value={this.state.birthday} ??? Destruct if if it's correct.

**Review by Gorett — APPROVED** [review]

File: src/App.js [hunk 19]
```diff
@@ -1,29 +1,18 @@
...
 
+class App extends Component {
+  
   render() {
     return (
       <MuiThemeProvider>
-        <div>
-          <Form onChange={fields => this.onChange(fields)} />
-        </div>
+        <Fragment>
```
Comment (Gorett) [inline src/App.js]: Try without Fragment here

**Review by Gorett — CHANGES_REQUESTED** [review]

File: src/components/Form.js (hunk 1 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/Form.js (hunk 2 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/Form.js (hunk 3 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/Form.js (hunk 4 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/Form.js (hunk 5 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/Form.js (hunk 6 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/Home.js (hunk 8 above)
Comment (NOXY1) [reply src/components/Home.js]: done

File: src/components/LogIn.js (hunk 9 above)
Comment (NOXY1) [reply src/components/LogIn.js]: done

File: src/components/LogIn.js (hunk 10 above)
Comment (NOXY1) [reply src/components/LogIn.js]: done

File: src/components/LogIn.js (hunk 11 above)
Comment (NOXY1) [reply src/components/LogIn.js]: done

File: src/components/NotFound.js (hunk 12 above)
Comment (NOXY1) [reply src/components/NotFound.js]: done

File: src/components/SignUp.js (hunk 13 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 14 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 15 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 16 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 17 above)
Comment (NOXY1) [reply src/components/SignUp.js]: ))))good joke

File: src/App.js (hunk 19 above)
Comment (NOXY1) [reply src/App.js]: done

File: src/components/SignUp.js [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+			            />
+			          }
+			          label="I accept the terms of the license agreement"
+			        />
+			        <br />
+			        <Button variant='contained'
+							label='Sign Up' 
+							onClick={e => this.handleSubmit(e)} 
+							color="primary"
+							disabled = { !isSignInBtnActive }
```
Comment (Gorett) [inline src/components/SignUp.js]: remove spaces here

File: src/components/Form.js [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+		}
+
+		return(
+			<Fragment>
+				<p className={ this.state.notificationType}>{this.state.notification }</p>
+				<form className='ml'>
+					<TextField 
+						name='login'
+						label='Login'
+						value={ login }
```
Comment (Gorett) [inline src/components/Form.js]: remove spaces here

File: src/components/Form.js [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='password'
+						label='Password'
+						value={ password }
```
Comment (Gorett) [inline src/components/Form.js]: remove spaces here

File: src/components/Form.js [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+						type='password'
+					/>
+					<br />
+					<Button variant='contained'
+							label='Submit' 
+							onClick={e => this.handleSubmit(e)} 
+							color="primary"
+							disabled={ !isLoginBtnActive }>
```
Comment (Gorett) [inline src/components/Form.js]: remove spaces here

File: src/components/LogIn.js [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    return(
+      <Fragment>
+        {!!notification && <p className={ notificationType }>{ notification }</p>}
+        <form className='ml'>
+          <TextField 
+            name='login'
+            label='Login'
+            value={ login }
```
Comment (Gorett) [inline src/components/LogIn.js]: remove spaces here

File: src/components/LogIn.js [hunk 25]
```diff
@@ -0,0 +1,83 @@
...
+            onChange={e => this.handleChangeInput(e)}
+            margin="normal"
+          />
+          <br />
+          <TextField 
+            name='password'
+            label='Password'
+            value={ password }
```
Comment (Gorett) [inline src/components/LogIn.js]: remove spaces here

File: src/components/LogIn.js [hunk 26]
```diff
@@ -0,0 +1,83 @@
...
+            onChange={e => this.handleChangeInput(e)} 
+            margin="normal"
+            type='password'
+          />
+          <br />
+          <Button variant='contained'
+              label='Submit' 
+              onClick={e => this.handleSubmit(e)} 
+              color="primary"
+              disabled={ !isLoginBtnActive }>
```
Comment (Gorett) [inline src/components/LogIn.js]: remove spaces here

File: src/components/SignUp.js [hunk 27]
```diff
@@ -0,0 +1,115 @@
+import React, { Fragment, Component } from 'react';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import Checkbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { signUp } from '../services/userService';
+import { Redirect } from 'react-router-dom';
+import '../App.css';
+
```
Comment (Gorett) [inline src/components/SignUp.js:9]: remove one empty line here

File: src/components/SignUp.js [hunk 28]
```diff
@@ -0,0 +1,115 @@
...
+						label='Surname'
+						value={ lastName }
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='email'
+						label='Email'
+						value={ email }
```
Comment (Gorett) [inline src/components/SignUp.js]: remove spaces here

File: src/components/SignUp.js [hunk 29]
```diff
@@ -0,0 +1,115 @@
...
+						label='Name'
+						value={ firstName }
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='lastName'
+						label='Surname'
+						value={ lastName }
```
Comment (Gorett) [inline src/components/SignUp.js]: remove spaces here

File: src/components/SignUp.js [hunk 30]
```diff
@@ -0,0 +1,115 @@
...
+		}
+
+		return(
+			<Fragment>
+			{!!notification && <p className={ notificationType }>{ notification }</p>}
+				<form className='ml'>
+					<TextField 
+						name='firstName'
+						label='Name'
+						value={ firstName }
```
Comment (Gorett) [inline src/components/SignUp.js]: remove spaces here

File: src/components/SignUp.js [hunk 31]
```diff
@@ -0,0 +1,115 @@
...
+						label='Email'
+						value={ email }
+						onChange={e => this.handleChangeInput(e)}
+						margin="normal"
+					/>
+					<br />
+					<TextField 
+						name='birthday'
+						label='Birthday'
+						value={ birthday }
```
Comment (Gorett) [inline src/components/SignUp.js]: remove spaces here

File: src/components/SignUp.js [hunk 32]
```diff
@@ -0,0 +1,115 @@
...
+		const { firstName, lastName, email, birthday, notification, notificationType } = this.state;
+		const isSignInBtnActive = !!(firstName && lastName && email && birthday);
+		
+		if(this.state.redirect) {
+			return (<Redirect to={ '/home' } />)
+		}
+
+		return(
+			<Fragment>
+			{!!notification && <p className={ notificationType }>{ notification }</p>}
```
Comment (Gorett) [inline src/components/SignUp.js]: remove spaces here

**Review by Gorett — APPROVED** [review]

File: src/components/SignUp.js (hunk 20 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/Form.js (hunk 21 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/Form.js (hunk 23 above)
Comment (NOXY1) [reply src/components/Form.js]: done

File: src/components/LogIn.js (hunk 24 above)
Comment (NOXY1) [reply src/components/LogIn.js]: done

File: src/components/SignUp.js (hunk 28 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 29 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 30 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 31 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

File: src/components/SignUp.js (hunk 32 above)
Comment (NOXY1) [reply src/components/SignUp.js]: done

**Pull request merged**